### ISSUE TITLE
Remove old database.py workarounds

### ISSUE THREAD

**Issue by qstokkink** [Tribler/py-ipv8#982]
The `database.py` file is **really old** and includes two workarounds we no longer need.

### Workaround for MacOS Sierra

https://github.com/Tribler/py-ipv8/blob/d95b22344b0adac84619c039366fba7eb4f2b043/ipv8/database.py#L16-L35

This was fixed in Python 3.7 - which is our current minimum version. This can be replaced with `import sqlite3 ` and `cursor.execute(statement)` for all instances of `execute_or_script`.

### Python 2 compatibility

https://github.com/Tribler/py-ipv8/blob/d95b22344b0adac84619c039366fba7eb4f2b043/ipv8/database.py#L38-L39

`database_blob()` has now been reduced to `bytes()`: it only adds needless complexity and should be replaced with `bytes()` everywhere.

**Comment by devos50** [comment]
Agreed, that file is very old. One thing that bothers me also is that the database forces a `sqlite` directory in the path. This should be quite easy to fix 👍 

From my side, only TrustChain and AnyDex are critically depending on the IPv8 database. I can easily fix compatibility with any change you decide on.

**Comment by qstokkink** [comment]
> One thing that bothers me also is that the database forces a `sqlite` directory in the path. This should be quite easy to fix 👍

That shouldn't be the case.

This seems to be [implemented seperately](https://github.com/Tribler/anydex-core/blob/8177d782714900613e3be74670ceb400aca36643/anydex/trustchain/database.py#L32) in `class TrustChainDB(Database)` in AnyDex:

```python
DATABASE_DIRECTORY = os.path.join("sqlite")

 ...

db_path = os.path.join(working_directory, os.path.join(DATABASE_DIRECTORY, u"%s.db" % db_name))
```

You should be able to fix this on the AnyDex side by removing the `DATABASE_DIRECTORY` prefix.

> I can easily fix compatibility with any change you decide on.

Nice 👍 You can preempt the removal of `database_blob` without depending on IPv8 (find and replace all instances of `database_blob` with `bytes`). The other change is tied to the IPv8 pointer.

**Comment by devos50** [comment]
Thanks! I remembered that was related to the IPv8 database but the issue seems to be local to AnyDex then indeed. I'll fix it later 👍

**Comment by devos50** [comment]
I fixed the `sqlite` path component and removed `database_blob` in [this PR](https://github.com/Tribler/anydex-core/pull/71) 👍

**Comment by qstokkink** [comment]
Thanks for the heads-up. That "only" leaves Tribler (which has a surprising amount of `database_blob`'s: https://github.com/Tribler/tribler/search?q=database_blob).

**Comment by qstokkink** [comment]
I'll put this issue on hold until https://github.com/Tribler/tribler/issues/6097 is resolved.

**Comment by qstokkink** [comment]
With the merge of https://github.com/Tribler/tribler/pull/6196, this enhancement is no longer on hold.
